src/sagas/repository.ts

```diff
@@ -909,6 +916,26 @@ export function* changePage(page: number): any {
   }
 }
 
+/**
+ * DUOS
+ */
+export function* getDuosDatasets(): any {
+  try {
+    const duosUrl = yield select(getDuosUrl);
+    const url = `${duosUrl}/api/dataset/v2`;
+    const response = yield call(authGet, url);
+    yield put({
+      type: ActionTypes.GET_DUOS_DATASETS_SUCCESS,
+      datasets: { data: response },
+    });
+  } catch (err) {
+    yield put({
+      type: ActionTypes.GET_DUOS_DATASETS_FAILURE,
+    });
+    showNotification(err);
+  }
```

Comment:
When I navigate to the snapshot page, I'm getting a 404 error. I think it's because I'm not registered in DUOS. 
I don't think we can expect that every TDR user will also be registered in DUOS. 

`GET https://consent.dsde-dev.broadinstitute.org/api/dataset/v2 404`
![image](https://github.com/DataBiosphere/jade-data-repo-ui/assets/13254229/7a1f1d3a-c9c3-44c6-9af6-1b672c31957d)

Reply:
oooh interesting.  Good catch.  I'll add something to handle that gracefully.  Perhaps I'll just grey out the drop down in that case (since only users with access to DUOS should be linking DUOS datasets

Reply:
ok, should be good to go!